src/liballoc/vec.rs

```diff
@@ -2609,6 +2799,22 @@ impl<T> FusedIterator for IntoIter<T> {}
 #[unstable(feature = "trusted_len", issue = "37572")]
 unsafe impl<T> TrustedLen for IntoIter<T> {}
 
+#[doc(hidden)]
+#[unstable(issue = "none", feature = "std_internals")]
+// T: Copy as approximation for !Drop since get_unchecked does not advance self.ptr
+// and thus we can't implement drop-handling
+unsafe impl<T> TrustedRandomAccess for IntoIter<T>
```

Comment:
Is this related to the FromIterator specialization change?

Reply:
as explained in the other comment, it's there to make in place iteration on `Zip` perform well.